tests/functional/test_interaction.py

```diff
@@ -0,0 +1,29 @@
+import os
+import mock
+import shutil
+import pytest
+import tempfile
+
+from awsshell.interaction import FilePrompt
+
+
+def test_file_prompt_execute():
+    # Test that the file propmpt correctly reads files
+    temp_dir = tempfile.mkdtemp()
+    temp_file = os.path.join(temp_dir + 'tmp')
+    try:
+        with open(temp_file, 'w') as f:
+            f.write('contents')
+
+        prompt = mock.Mock()
+        file_prompt = FilePrompt({}, 'msg', prompt)
+        prompt.side_effect = [temp_file, temp_dir + 'notafile']
+
+        result = file_prompt.execute({})
+        assert result == 'contents'
+
+        # IOError for Python2 compatibility, 3 throws FileNotFoundError
+        with pytest.raises(IOError):
```

Comment:
I did this to avoid having to set up the filesystem twice. Perhaps a bit of a premature optimization. I'll split this into two tests using a more proper setup/teardown structure.